docs/integrations/github/discovery.md

```diff
@@ -80,6 +80,21 @@ catalog:
         filters: # optional filters
           branch: 'develop' # optional string
           repository: '.*' # optional Regex
+      topicProviderId:
+        organization: 'backstage' # string
+        catalogPath: '/catalog-info.yaml' # string
+        filters:
+          branch: 'main' # string
+          repository: '.*' # Regex
+          topic: 'backstage-exclude' # optional string
+      topicProviderInclusionId:
+        organization: 'backstage' # string
+        catalogPath: '/catalog-info.yaml' # string
+        filters:
+          branch: 'main' # string
+          repository: '.*' # Regex
+          topic: 'backstage-include' # optional string
+          topicIncludesIfMatch: true # optional boolean (default: false)
```

Comment:
given both branch and repository are inclusion filters it seems odd to be that `topic` defaults to exclude.

it could be more explicit with something like:

```yaml
filters:
  topicInclude: `backstage-include`
  topicExclude: `backstage-exclude`
```

or
```yaml
filters:
  topic:
    name: `backstage-include
    type: `INCLUDES`
```

Reply:
Ah yeah, that is true so I can invert it.

I did consider using an enum like `INCLUDES` but I don't really know what other filters there would be so I just modelled it as a boolean since it was a bit easier than setting up an enum.

Going more explicit could be a path although it involves handing what to do when a user inevitably tried to provide both an include and exclude at the same time. I guess you could just throw an error and tell the user to pick one or the other.

Reply:
I was thinking about it a bit today and I think overall, having an enum is probably clearer and requires less thought

Reply:
@timja I ended up adopting your second suggestion which I think is less confusing as to how it'll behave overall. I've also updated the default to be exclusion 🙂